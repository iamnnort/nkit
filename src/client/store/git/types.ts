import { CommonStore } from '../../lib/store/types';

export enum ActionTypes {
  LOAD_RELEASES_REQUEST = '@@git/LOAD_RELEASES_REQUEST',
  LOAD_RELEASES_SUCCESS = '@@git/LOAD_RELEASES_SUCCESS',
  LOAD_RELEASES_FAILURE = '@@git/LOAD_RELEASES_FAILURE',
}

export interface Git extends CommonStore {
  releases: GitRelease[];
}

export interface GitRelease {
  id: number;
  name: string;
  tag_name: string;
  html_url: string;
}
