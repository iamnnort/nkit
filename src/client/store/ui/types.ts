export enum ActionTypes {
  TOGGLE_SIDEBAR = '@@ui/TOGGLE_SIDEBAR',
}

export interface UiInterface {
  isSidebarOpened: boolean;
}
