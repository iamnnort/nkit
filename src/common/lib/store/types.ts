export interface CommonStore {
  isInitialLoaded: boolean;
  isLoading: boolean;
  error: string;
}

export enum CommonError {
  FAILED_REQUEST = 'Loading was failed',
}
