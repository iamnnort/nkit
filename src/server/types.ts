import * as express from 'express';
import i18next from 'i18next';

export interface Request extends express.Request {
  i18n: i18next.i18n;
}
