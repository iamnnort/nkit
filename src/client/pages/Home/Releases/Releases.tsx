import * as React from 'react';

import { Releases, ReleasesTitle, ReleasesList, ReleasesListItem, ReleasesListItemLink } from './Releases.styled';
import { ComponentProps } from './Releases.types';
import { Icon } from '../../../components';

const ReleasesComponent: React.FC<ComponentProps> = ({ releases }) => {
  return (
    <Releases>
      <ReleasesTitle>{'Releases'}</ReleasesTitle>
      <ReleasesList>
        {releases.map(release => (
          <ReleasesListItem key={release.id}>
            <Icon type={'tag'} />
            <ReleasesListItemLink href={release.html_url} target={'_blank'}>
              {`${release.tag_name} (${release.name})`}
            </ReleasesListItemLink>
          </ReleasesListItem>
        ))}
      </ReleasesList>
    </Releases>
  );
};

export default ReleasesComponent;
