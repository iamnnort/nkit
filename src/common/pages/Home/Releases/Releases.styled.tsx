import styled from '@common/theme/styled-components';

export const Releases = styled.section``;

export const ReleasesTitle = styled.div`
  font-size: ${(props) => props.theme.mediumFontSize};
  font-weight: bold;
`;

export const ReleasesList = styled.div`
  margin-top: ${(props) => props.theme.smallPadding};
  padding-left: 5px;
`;

export const ReleasesListItem = styled.div`
  display: flex;
  align-items: center;

  svg {
    margin-right: 10px;
  }
`;

export const ReleasesListItemLink = styled.a``;
