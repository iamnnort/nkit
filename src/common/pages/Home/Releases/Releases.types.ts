import { GitRelease } from '@common/store/git/types';

export interface ComponentProps {
  releases: GitRelease[];
}
