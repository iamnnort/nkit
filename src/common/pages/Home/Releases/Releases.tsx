import * as React from 'react';
import { useTranslation } from 'react-i18next';

import {
  Releases,
  ReleasesTitle,
  ReleasesList,
  ReleasesListItem,
  ReleasesListItemLink,
} from '@common/pages/Home/Releases/Releases.styled';
import { ComponentProps } from '@common/pages/Home/Releases/Releases.types';
import { Icon } from '@common/components';

const ReleasesComponent: React.FC<ComponentProps> = ({ releases }) => {
  const { t } = useTranslation();

  return (
    <Releases>
      <ReleasesTitle>{t('general:releases')}</ReleasesTitle>
      <ReleasesList>
        {releases.map((release) => (
          <ReleasesListItem key={release.id}>
            <Icon type={'tag'} />
            <ReleasesListItemLink href={release.html_url} target={'_blank'}>
              {`${release.tag_name} (${release.name})`}
            </ReleasesListItemLink>
          </ReleasesListItem>
        ))}
      </ReleasesList>
    </Releases>
  );
};

export default ReleasesComponent;
