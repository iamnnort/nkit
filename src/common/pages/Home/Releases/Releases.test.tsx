import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import ReleasesComponent from '@common/pages/Home/Releases/Releases';
import { ComponentProps } from '@common/pages/Home/Releases/Releases.types';

jest.mock('../../../components/Icon/Icon', () => global.mockComponent('Icon'));

describe('ReleasesComponent', () => {
  let wrapper: ReactWrapper;

  const props: ComponentProps = {
    releases: [
      {
        id: 1,
        name: 'Fake name',
        tag_name: 'fake_name',
        html_url: 'fake_url',
      },
    ],
  };

  beforeEach(() => {
    wrapper = mount(<ReleasesComponent {...props} />);
  });

  it(`should not regret`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
