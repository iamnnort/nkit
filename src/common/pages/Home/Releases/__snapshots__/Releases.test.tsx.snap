// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleasesComponent should not regret 1`] = `
.c0 {
  font-weight: bold;
}

.c1 {
  padding-left: 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 svg {
  margin-right: 10px;
}

<ReleasesComponent
  releases={
    Array [
      Object {
        "html_url": "fake_url",
        "id": 1,
        "name": "Fake name",
        "tag_name": "fake_name",
      },
    ]
  }
>
  <styled.section>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bDWFJH",
            "rules": Array [],
          },
          "displayName": "styled.section",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className=""
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c0",
                  "rules": Array [
                    "
  font-size: ",
                    [Function],
                    ";
  font-weight: bold;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c0"
            >
              general:releases
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  margin-top: ",
                    [Function],
                    ";
  padding-left: 5px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              <styled.div
                key="1"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
  display: flex;
  align-items: center;

  svg {
    margin-right: 10px;
  }
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="c2"
                  >
                    <default
                      type="tag"
                    >
                      <MockComponent
                        originalComponent="Icon"
                        type="tag"
                      />
                    </default>
                    <styled.a
                      href="fake_url"
                      target="_blank"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-ifAKCX",
                              "isStatic": false,
                              "lastClassName": "cFlEyZ",
                              "rules": Array [],
                            },
                            "displayName": "styled.a",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-ifAKCX",
                            "target": "a",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        href="fake_url"
                        target="_blank"
                      >
                        <a
                          className=""
                          href="fake_url"
                          target="_blank"
                        >
                          fake_name (Fake name)
                        </a>
                      </StyledComponent>
                    </styled.a>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
      </section>
    </StyledComponent>
  </styled.section>
</ReleasesComponent>
`;
